cr.js

```diff
@@ -35,26 +46,78 @@ PriorityQueue.prototype.get_all_priorities = function() {
 
 // iterates through all the queue elements in priority-then-FIFO order
 PriorityQueue.prototype.forEach = function(callback) {
-    var keys = Object.keys(this.store).sort();
 
-    for (var a = keys.length; a > 0; a--) {
-        for (var b = 0; b < this.store[a].length; b++)
-            callback(this.store[a][b]);
+    for (var key in this.store) {
+        this.store[key].forEach(function(val, indexOfVal){
+            callback(key, val, indexOfVal)
+        });
     }
 }
 
 PriorityQueue.prototype.changePriority = function(value, newPriority) {
-    var foundItem = false;
-
-    this.store.forEach(function(bucket) {
-        bucket.forEach(function(item, index) {
-            if (item === value) {
-                bucket.splice(index, 1);  // remove the item
-                this.add(value, newPriority);
-                foundItem = true;
-                return false;  // early exit from forEach
-            }
-        });
-        if (foundItem) return false;
+
+    self = this;
+    foundItem = false;
+
+    this.forEach(function(key, val, indexOfVal){
+        if (value === val) {
+            self.store[key].splice(indexOfVal, 1);
+            self.count--;
+            self.add(value, newPriority);
+            foundItem = true;
+        }
     });
-}
```

Comment:
this.store.forEach was complaining, could have iterated through keys and then through values to get to the needed value, but thought we already have the forEach implemented as part of the prototype, so why not use it. While removing the value first, we also need to decrement the count, the add method will increment it back it when it assigns that value a new priority.